crates/types/src/journal_v2/raw/mod.rs

```diff
@@ -54,18 +54,39 @@ pub enum RawEntryError {
     TryFromEntry(#[from] TryFromEntryError),
 }
 
+// todo split RawEntryHeader into headers for Bifrost (values that need to be agreed upon) and
+//  headers for the partition store (other values + values that can be derived from the Bifrost
+//  record (e.g. the append_time))
```

Comment:
imo what makes sense to do here would be to have the invoker effect take only raw entry inner, and then eventually addittioinal metadata gets added later directly in the invoker effect. you can probably do this even now, without modifying this data structure...